src/variational/distributions.jl

```diff
@@ -0,0 +1,57 @@
+import Distributions: _rand!
+
+
+function jac_inv_transform(dist::Distribution, x::T where T<:Real)
+    ForwardDiff.derivative(x -> invlink(dist, x), x)
+end
+
+function jac_inv_transform(dist::Distribution, x::AbstractArray{T} where T <: Real)
+    ForwardDiff.jacobian(x -> invlink(dist, x), x)
+end
+
+function jac_inv_transform(dist::Distribution, x::TrackedArray{T} where T <: Real)
+    Tracker.jacobian(x -> invlink(dist, x), x)
+end
+
+function center_diag_gaussian(x, μ, σ)
+    # instead of creating a diagonal matrix, we just do elementwise multiplication
+    (σ .^(-1)) .* (x - μ)
+end
+
+function center_diag_gaussian_inv(η, μ, σ)
+    (η .* σ) + μ
+end
+
+# Mean-field approximation used by ADVI
+struct MeanField{T, TDists <: AbstractVector{<: Distribution}} <: VariationalPosterior where T <: Real
+    μ::Vector{T}
```

Comment:
Is there a particular reason to make the types here `Vector`s, rather than some subtype of `AbstractVector`?

Reply:
@torfjelde I found this one is not resolved. Can you quick fix it?

Reply:
Ah, cheers! I've addressed this now:)